executor/include/execution/executor.hpp

```diff
@@ -40,43 +40,40 @@ TEST_CASE_TEMPLATE_DEFINE("Validity ", E, validity) {
 }
 
 TEST_CASE_TEMPLATE_DEFINE("Property Traits ", E, property_traits) {
-  const auto exec = E{};
+  auto exec = E{};
 
   SUBCASE("can_require") {
-    CHECK(execution::can_require_v<decltype(exec), blocking_t::always_t> ==
-          true);
-    CHECK(execution::can_require_v<decltype(exec), blocking_t::never_t> ==
-          false);
+    CHECK(execution::can_require_v<E, blocking_t::always_t> == true);
+    CHECK(execution::can_require_v<E, blocking_t::never_t> == false);
   }
 
   SUBCASE("can_prefer") {
-    CHECK(execution::can_prefer_v<decltype(exec), blocking_t::always_t> ==
-          true);
-    CHECK(execution::can_prefer_v<decltype(exec), blocking_t::never_t> == true);
+    CHECK(execution::can_prefer_v<E, blocking_t::always_t> == true);
+    CHECK(execution::can_prefer_v<E, blocking_t::never_t> == true);
   }
 }
 
 TEST_CASE_TEMPLATE_DEFINE("Properties", E, properties) {
-  const auto exec = E{};
+  auto exec = E{};
 
   SUBCASE("member require & query") {
     CHECK(exec.query(blocking.never) == false);
-    auto new_exec =
-        static_cast<E>(exec).require(oneway).require(blocking.always);
+    auto new_exec = typename std::remove_cv<E>::type{}.require(oneway).require(
```

Comment:
I restate my belief that `require` can be declared a const method.

Reply:
Yeah, no modification taking place.